File: AlgorithmsCpp/799_最长连续不重复子序列.cpp
# include<iostream>

using namespace std;

const int N = 100010;
int n;
int a[N], s[N];

int main() {
	cin >> n;
	for (int i = 0; i < n; i++) cin >> a[i];

	int result = 0;
	for (int i = 0, j = 0; i < n; i++) {
		s[a[i]]++;
		while (s[a[i]] > 1) s[a[j]]--, j++;
		result = max(result, i - j + 1);
	}

	printf("%d", result);
	return 0;
}

File: AlgorithmsCSharp/基础算法/799_最长连续不重复子序列.cs
namespace _799_最长连续不重复子序列
{
    class Program
    {
        static void Main()
        {
            int n = Convert.ToInt32(Console.ReadLine());
            int[] a = new int[n];
            int[] s = new int[100010];
            for (int i = 0; i < n; i++)
            {
                a[i] = Convert.ToInt32(Console.ReadLine());
            }

            int result = 0;
            for (int i = 0, j = 0; i < n; i++)
            {
                s[a[i]]++;
                while (s[a[i]] > 1)
                {
                    s[a[j]]--;
                    j++;
                }

                result = Math.Max(result, i - j + 1);
            }

            Console.WriteLine(result);
        }
    }
}
